v2x_ros_driver/src/v2x_radio_client.cpp

```diff
@@ -168,29 +169,124 @@ void V2XRadioClient::process(const std::shared_ptr<const std::vector<uint8_t>>&
         else {
             // TODO lengths greater than 16383 (0x3FFF) are encoded by splitting up the message into discrete chunks, each with its own length
             // marker. It doesn't look like we'll be receiving anything that long
-            std::cerr << "V2XRadioClient::process() : received a message with length field longer than 16383." << std::endl;
+            RCLCPP_WARN_STREAM(logger_, "V2XRadioClient::process() : discarding received message with length field longer than 16383.");
             continue;
         }
         if (len == -1) { continue; }
         // If the length makes sense bsmPub(fits in the buffer), copy out the message bytes and pass to the Application class
         if ((i + 1 + len + len_bytes) < entry.size()) {
-            // bool found_valid_msg = true;
+            // First step in checking for valid message. Message must begin with a valid MessageID. 
             if (!IsValidMsgID(std::to_string(msg_id))) { continue; }
+
             size_t start_index = i;
             size_t end_index = i + 2 + len + len_bytes; // includes 2 msgID bytes before message body
-            //this constructor has range [first, last) hence the + 1
+            // this constructor has range [first, last) hence the + 1
             std::vector<uint8_t> msg_vec(entry.begin() + start_index, entry.begin() + end_index);
-            onMessageReceived(msg_vec, msg_id);
-            break;
+
+            // Second check for a valid message. Checks MessageFrame length field against actual payload size.
+            if (!isValidMsgSize(msg_vec, start_index, end_index, entry))
+            {
+                RCLCPP_WARN_STREAM(logger_, "Size in possible MessageFrame does not match actual data size. Checking rest of data.");
+                continue;
+            }
+            else
+            {
+                // Last check for a valid message. WSA uses same format as WAVE Short Message (WSM) and precurses the WSM in some cases.
+                // Make sure WSA is not accidentally detected before actual message. This is done by checking detected msg_id against list of PSIDs.
+                if (!isValidPSID(std::to_string(msg_id)))
+                {
+                    printVectorHelper(msg_vec);
```

Comment:
Do you think `printVectorHelper()` is still needed since `OnMessageReceived` already prints the vector to RCLCPP_DEBUG_STREAM ([see here](https://github.com/usdot-fhwa-stol/v2x-ros-driver/blob/58cc0cf3c0ee0ead9ecd7e1b845f8799a57d4b56/v2x_ros_driver/src/v2x_ros_driver_node.cpp#L155C115-L155C141)). Or do `printVectorHelper()` and `uint8_vector_to_hex_string()` format the vector differently?

Reply:
They would be in the same format, but printVectorHelper would will only print out the valid message vs OnMessageReceived printing out all received messages. Think we should leave this in?

Reply:
Removed helper function